common/package.json

```diff
@@ -1,36 +1,29 @@
 {
-  "name": "xtypescript",
+  "name": "@exercism/typescript",
   "version": "1.0.0",
   "description": "Exercism exercises in Typescript.",
   "author": "",
   "private": true,
   "repository": {
     "type": "git",
-    "url": "https://github.com/exercism/xtypescript"
+    "url": "https://github.com/exercism/typescript"
   },
-  "devDependencies": {},
-  "scripts": {
-    "test": "tsc --noEmit -p . && jest --no-cache",
-    "lint": "tsc --noEmit -p . && tslint \"*.ts?(x)\"",
-    "lintci": "tslint \"*.ts?(x)\" --force"
+  "devDependencies": {
+    "@types/jest": "^24.0.15",
+    "@types/node": "^12.6.8",
+    "@typescript-eslint/eslint-plugin": "^1.13.0",
+    "@typescript-eslint/parser": "^1.13.0",
+    "eslint": "^6.1.0",
+    "eslint-plugin-import": "^2.18.2",
+    "jest": "^24.8.0",
+    "ts-jest": "^24.0.2",
+    "typescript": "^3.5.3"
   },
-  "dependencies": {
-    "@types/jest": "^21.1.5",
-    "@types/node": "^8.0.47",
-    "jest": "^21.2.1",
-    "ts-jest": "^21.1.3",
-    "tslint": "^5.8.0",
-    "typescript": "^2.5.3"
+  "scripts": {
+    "test": "yarn lint:types && jest --no-cache",
+    "lint": "yarn lint:types && yarn lint:ci",
```

Comment:
ci lint and user lint have different rules. It's just named that by convention.

Reply:
(If you think the name should be changed because it's confusing, please provide a suggestion 👍. I'm not married to the name 😄 )

Reply:
Instead of doing the magic of different lint rule checking in the make file, you can define that work away by just having two distinct lint commands. One that gets run for on the CI only which would have a non standard configuration file name and the other will be the the default standard name config which would be the default.

Reply:
I absolutely no longer understand what you mean.

`common/package.json` has `lint` which will run two linters: compile linting (via tsc, called lint:types, but it's just a name) and eslint linting (called lint:ci, but it's just a name). Both allow for watching, and this allows students to watch on either.

- They can watch on `tsc` if they don't use a supported IDE that does it for them, whilst writing code
- They can watch on `eslint` once they are ready to submit
- They can choose not to lint at all, all of this is optional

When linting by a student, it will use `common/.eslintrc`.

----

`maintaining/.eslint.track.rc` is the eslint configuration that is used by the CI/us maintainers/the make file. Because everything else is equal (it still needs to check types, which uses `tsconfig.json` , which we can also provide separately (read: stricter version for ci/make/maintainers), it can re-use `yarn lint` which will call `yarn lint:types` and `yarn lint:ci`.

Reply:
What you call lint:types is not a linter. That is the typescript type checker. That one needs a different name. Maybe typechecker. We needed because jest doesn’t types check so a test can fail because of a type error. 

The lint was meant to be used by users but previously the rules were the same

lintci was different because we don’t want the lint to error out on the ci but we do want it to report the error so danger can pick them up and report on PRs. 

So now we are introducing a ESLint. So we need eslint to create the violation json file when running on the ci (lintci)    So now we have to parse that json so danger picks them up. I don’t thinking fixing those violation should be part of this pr. 

The regular lint command (lint) we don’t want the json file and since this is running on the local dev machine it should error or warn which the ide would pick up.

Reply:
I think we're partially on the same wave length and partially completely missing each other's points/concerns. 😓 

> We needed because jest doesn’t types check so a test can fail because of a type error.

ts-jest doest, but I understand what you're saying.

> lintci was different because we don’t want the lint to error out on the ci but we do want it to report the error so danger can pick them up and report on PRs.

No, that's not what it was. The "lintci" you're talking about was an extra, hardcoded, `make` rule that built the lintreport.json (but ignored errors). It never used the old `package.json` `lintci` to build the danger report. 

> So now we are introducing a ESLint. So we need eslint to create the violation json file when running on the ci (lintci) So now we have to parse that json so danger picks them up. I don’t thinking fixing those violation should be part of this pr.

This is in `package.track.json` in maintainers and called `report`. I already parse it (already updated danger to work with eslint) and it already works (see edit history on the pacman bot message). If we remove these changes from this PR, danger breaks, and the makefile breaks, because both were hardcoding and relying on `tslint`.

> The regular lint command (lint) we don’t want the json file and since this is running on the local dev machine it should error or warn which the ide would pick up.

Correct, the `common/package.json` lint:ci always errors, and the IDE will pick it up. It doesn't generate a json file.